lib/config/config.js

```diff
@@ -305,7 +536,11 @@ class Config {
 				: [originalConfig];
 
 			// normalize severity
-			ruleConfig[0] = severities.get(ruleConfig[0]);
+			ruleConfig[0] = severities.get(
+				typeof ruleConfig[0] === "string"
+					? ruleConfig[0].toLowerCase()
+					: ruleConfig[0],
+			);
```

Comment:
I think this change is unnecessary because at this point the value can only be `0`, `1`, `2`, `"off"`, `"warn"`, or `"error"` as the severity is already validated in flat-config-schema:

https://github.com/eslint/eslint/blob/60c3e2cf9256f3676b7934e26ff178aaf19c9e97/lib/config/flat-config-schema.js#L508-L533